src/components/context/CodeVersionContext.tsx

```diff
@@ -0,0 +1,120 @@
+import { List } from 'immutable';
+
+import React, {
+  FC,
+  PropsWithChildren,
+  createContext,
+  useMemo,
+  useState,
+} from 'react';
+
+import { INSTRUCTOR_CODE_VERSION_SETTINGS_KEY } from '../../config/appSettingsTypes';
+import { DEFAULT_CODE_VERSION_SETTING } from '../../config/codeVersions';
+import {
+  ANONYMOUS_USER,
+  INSTRUCTOR_CODE_ID,
+  INSTRUCTOR_CODE_NAME,
+} from '../../config/constants';
+import { hooks } from '../../config/queryClient';
+import {
+  CodeVersionSelectType,
+  CodeVersionType,
+} from '../../interfaces/codeVersions';
+import { useAppDataContext } from './AppDataContext';
+import { useMembersContext } from './MembersContext';
+
+type CodeVersionContextType = {
+  codeVersionResource: CodeVersionSelectType;
+  codeVersion: CodeVersionType;
+  groupedVersions: {
+    user: { value: string; label: string };
+    versions: CodeVersionSelectType[];
+  }[];
+  codeId: string; // id of the appData with the current code
+  setCodeId: (id: string) => void;
+};
+
+const defaultCodeVersion = {
+  id: INSTRUCTOR_CODE_ID,
+  data: DEFAULT_CODE_VERSION_SETTING,
+  creator: INSTRUCTOR_CODE_NAME,
+  updatedAt: '',
```

Comment:
I think you defined a date constant when this doesn't exist? You could reuse it here.

Reply:
The date constant i defined was used to define the text to display when there is no date. So here i set an empty string, such that the displayed value will be the constant i defined ... Does that make sense ?